src/bthread/mutex.h

```diff
@@ -48,15 +48,15 @@ class Mutex {
     Mutex() {
         int ec = bthread_mutex_init(&_mutex, NULL);
         if (ec != 0) {
-            throw std::system_error(std::error_code(ec, std::system_category()), "Mutex constructor failed");
+            CHECK(false) << "Mutex constructor failed";
         }
     }
     ~Mutex() { CHECK_EQ(0, bthread_mutex_destroy(&_mutex)); }
     native_handler_type native_handler() { return &_mutex; }
     void lock() {
         int ec = bthread_mutex_lock(&_mutex);
         if (ec != 0) {
-            throw std::system_error(std::error_code(ec, std::system_category()), "Mutex lock failed");
```

Comment:
Hi @x2c3z4 , Thanks for your contribution! But I have a small question, why remove the exception here, it was been added by #645

Reply:
Google C++ code style suggests that don't use the c++ exception, and I think a lot of projects have disabled the c++ exception including mine. All the brpc code is worked with c++ no-exception, except this code snippet.

If we don't accept this error, just abort it.
As I know, if the mutex has been freed or is a bad memory address, the pthread_mutex_lock will return EINVAL(22 Invalid argument). CHECK(false) is better than c++ exception.

Reply:
@x2c3z4 Google Style is good but I guess there is no way to ensure no one else is using this behavior. Removing the exception may break others codebase.

Reply:
What codebase will it break? I don't understand why needs to throw exception here.